apisix/core/config_etcd.lua

```diff
@@ -644,6 +662,7 @@ local function sync_data(self)
 
     if not dir_res then
         if err == "compacted" or err == "restarted" then
+            sync_status_to_shdict(self.need_reload)
```

Comment:
In this scenario, the status should be false, right?

Reply:
yes it will be false. you can look at the logic on status_ready() handler

Reply:
done